src/controller_examples/rabbitmq_controller/spec/reconciler.rs

```diff
@@ -523,26 +543,46 @@ pub open spec fn make_server_config_map(rabbitmq: RabbitmqClusterView) -> Config
         rabbitmq.metadata.name.is_Some(),
         rabbitmq.metadata.namespace.is_Some(),
 {
-    ConfigMapView::default()
-        .set_metadata(ObjectMetaView::default()
-            .set_name(make_server_config_map_name(rabbitmq.metadata.name.get_Some_0()))
-            .set_namespace(rabbitmq.metadata.namespace.get_Some_0())
-            .set_owner_references(seq![rabbitmq.controller_owner_ref()])
-            .set_labels(Map::empty().insert(new_strlit("app")@, rabbitmq.metadata.name.get_Some_0()))
-        )
-        .set_data(Map::empty()
-            .insert(new_strlit("operatorDefaults.conf")@, default_rbmq_config(rabbitmq))
-            .insert(new_strlit("userDefinedConfiguration.conf")@,
-            {
-                if rabbitmq.spec.rabbitmq_config.is_Some()
-                && rabbitmq.spec.rabbitmq_config.get_Some_0().additional_config.is_Some()
-                {   // check if there are rabbitmq-related customized configurations
-                    new_strlit("total_memory_available_override_value = 1717986919\n")@ + rabbitmq.spec.rabbitmq_config.get_Some_0().additional_config.get_Some_0()
-                } else {
-                    new_strlit("total_memory_available_override_value = 1717986919\n")@
-                }
-            })
-        )
+    ConfigMapView {
+        metadata: ObjectMetaView {
+            name: Some(make_server_config_map_name(rabbitmq.metadata.name.get_Some_0())),
+            namespace: rabbitmq.metadata.namespace,
+            owner_references: Some(seq![rabbitmq.controller_owner_ref()]),
+            labels: Some(make_labels(rabbitmq)),
+            annotations: Some(rabbitmq.spec.annotations),
+            ..ObjectMetaView::default()
+        },
+        data: Some({
+            let data = Map::empty()
+                        .insert(new_strlit("operatorDefaults.conf")@, default_rbmq_config(rabbitmq))
+                        .insert(new_strlit("userDefinedConfiguration.conf")@,
+                        {
+                            if rabbitmq.spec.rabbitmq_config.is_Some()
+                            && rabbitmq.spec.rabbitmq_config.get_Some_0().additional_config.is_Some()
+                            {   // check if there are rabbitmq-related customized configurations
+                                new_strlit("total_memory_available_override_value = 1717986919\n")@ + rabbitmq.spec.rabbitmq_config.get_Some_0().additional_config.get_Some_0()
```

Comment:
Interesting. Is this also hardcoded in the official controller? And if so, do you know why?

Reply:
Actually in the official controller, this key-value is inserted only when the cr specifies the memory limit (i.e., `cr.spec.resources.limits["memory"]`) and the limit will be the value. So it's conditional. 

But @ZichengMa set a default value here without any conditions. I never checked this before.

Reply:
The `limits` field of `ResourceRequirements` is added by me and also the `resources` field of cr was added recently, so we must not have had that memory limit before.

Reply:
Sure

Reply:
Just a note: to implement that feature you might need to call StringMap.get which is currently `pub fn get<'a>(&'a self, key: String) -> (v: Option<StrSlice<'a>>)` -- if this one does not work well for your case pls feel free to implement other getter method that directly returns Option<String>

Reply:
Another thing I want to ask is how to compare a `Quantity` value to zero? Should I write a parser function?

Reply:
Not only they require there is a memory limit in the resource field, but also the limit should not be zero.

Reply:
I think you can skip that limit !=0 check for now -- they are trying to ensure they put in a correct limit, but honestly speaking only checking whether it equals zero is far from enough -- e.g., 1KB is not zero, but does not sound like a correct limit. BTW, do they directly use the Quantity literal as `total_memory_available_override_value` or they need to covert it somehow (like the unites might be different)?

Reply:
They do a series of transitions which look non-trivial.

Reply:
But the methods they use are all provided by the `quantity` module of k8s api